docs/pbc/all/payment-service-provider/202410.0/base-shop/third-party-integrations/payone/app-composition-platform-integration/payone-acp-app.md

```diff
@@ -87,11 +87,169 @@ When customers pay with PayPal, a shop owner can do the following:
+
+
+### Example of the headless checkout with PayOne
+
+The PaymentMethods which are returned by the `/checkout-data?include=payment-methods` endpoint are used to render the express-checkout buttons e.g. on the Product Detail Page or the Cart Page. The customer can select the payment method and proceed to the payment page.
+
+You can read more details about this in the [Payment Method Strategies](https://documentation.spryker.com/docs/pbc/all/payment-service-provider/202410.0/base-shop/payment-method-strategies.html) document.
+
+When the custome clicks on "Pay with PayPal Express" button, the `InitializePreOrderPayment` Glue API endpoint is called to initialize the payment.
+
+#### Pre-order payment initialization
+
+```JS
+
+async initializePreOrderPayment() {
+    const requestData = {
+      data: {
+        type: 'payments',
+        attributes: {
+          quote: QUOTE_DATA,
+          payment: {
+            amount: GRAND_TOTAL, // You will get it through the `/checkout-data?include=carts` endpoint
+            paymentMethodName: 'PayPal Express', // taken from /checkout-data?include=payment-methods
+            paymentProviderName: 'PayOne',  // taken from /checkout-data?include=payment-methods
+          },
+          preOrderPaymentData: {
+            "transactionId": this.transactionId, // This is empty in the first request but has to be used in further requests
+          },
+        },
+      },
+    };
+
+    const responseData = await this.fetchHandler(`GLUE_APPLICATION_URL/payments`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ACCESS_TOKEN`,
+      },
+      body: JSON.stringify(requestData),
+    });
+
+    // This gets passed back from the App server and is different for each PSP
+    this.preOrderPaymentData = responseData.data.attributes.preOrderPaymentData;
+  
+    return this.preOrderPaymentData; // This has to be forwarded in any upcoming request
+  }
+
+```
+
+To identify the customer when initiating a request using Glue API, use the `Authorization` header for a logged-in customer and `X-Anonymous-Customer-Unique-Id` for a guest user.
+
+After making a request to the PayOne API, the payment is created in the PayOne app. The response looks as follows:
+
+```JSON
+{
+  "data": {
+    "type": "payments",
+    "attributes": {
+      "isSuccessful": true,
+      "error": null,
+      "preOrderPaymentData": {
+        "transactionId": "...",
+        "orderId": "...",
+        "workorderid": "...",
+        "token": "...",
+        "currency": "USD",
+        "merchantId": "..."
+      }
+    }
+  }
+}
+```
+
+After the customer clicks "Complete Purchase" in the PayPal Express modal he should be redirected to the summary page. 
+
+Since we skipped some important checkout steps here (addresses, shipment method, etc) another request needs tpo be made to get the customer data.
```

Comment:
```suggestion
Since we skipped some important checkout steps here (addresses, shipment method, etc) another request needs to be made to get the customer data.
```